src/celestia/celestiacore.cpp

```diff
@@ -2416,8 +2418,8 @@ void CelestiaCore::setFOVFromZoom()
                 v->observer->setFOV(MaximumFOVFisheye);
             else
             {
-                double fov = 2 * atan(height * v->height / (screenDpi / 25.4) / 2. / distanceToScreen) / v->zoom;
-                v->observer->setFOV((float) fov);
+                float fov = 2.0f * std::atan(static_cast<float>(height) * v->height / (static_cast<float>(screenDpi) / 25.4f) / 2.0f / static_cast<float>(distanceToScreen)) / v->getObserver()->getZoom();
```

Comment:
could you create a helper method to calculate `2.0f * std::atan(static_cast<float>(height) * v->height / (static_cast<float>(screenDpi) / 25.4f) / 2.0f / static_cast<float>(distanceToScreen))`?